src/net/http_serde.rs

```diff
@@ -109,10 +109,13 @@ pub mod query {
         async fn from_request(req: &mut RequestParts<B>) -> Result<Self, Self::Rejection> {
             #[derive(serde::Deserialize)]
             struct QueryTypeParam {
+                // TODO: serde custom error?
+                size: NonZeroU32,
```

Comment:
We might need a new type for this is `MAX_QUERY_SIZE` changes.

Reply:
added a new type `QuerySize` - made it cleaner